source/utilities.tex

```diff
@@ -15581,6 +15644,215 @@
 \end{itemdescr}
 
+\indexlibraryglobal{\exposid{range-default-formatter}}%
+\begin{codeblock}
+namespace std {
+  template<ranges::@\libconcept{input_range}@ R, class charT>
+  struct @\exposid{range-default-formatter}@<range_format::map, R, charT> {
+  private:
+    using @\exposid{maybe-const-map}@ = @\exposid{fmt-maybe-const}@<R, charT>;                        // \expos
+    using @\exposid{element-type}@ =                                                      // \expos
+      remove_cvref_t<ranges::range_reference_t<@\exposid{maybe-const-map}@>>;
+    range_formatter<@\exposid{element-type}@, charT> @\exposid{underlying_}@;                         // \expos
+  public:
+    constexpr formatter();
+
+    template<class ParseContext>
+      constexpr typename ParseContext::iterator
+        parse(ParseContext& ctx);
+
+    template<class FormatContext>
+      typename FormatContext::iterator
+        format(@\exposid{maybe-const-map}@& r, FormatContext& ctx) const;
+  };
+}
+\end{codeblock}
+
+\indexlibrarymember{formatter}{\exposid{range-default-formatter}}%
+\begin{itemdecl}
+constexpr formatter();
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\mandates
+Either:
+\begin{itemize}
+\item
+\exposid{element-type} is a specialization of \tcode{pair}, or
+\item
+\exposid{element-type} is a specialization of \tcode{tuple} and
+\tcode{tuple_size_v<T> == 2}.
+\end{itemize}
+
+\pnum
+\effects
+Equivalent to:
+\begin{codeblock}
+@\exposid{underlying_}@.set_brackets(@\exposid{STATICALLY-WIDEN}@<charT>("{"), @\exposid{STATICALLY-WIDEN}@<charT>("}"));
+@\exposid{underlying_}@.underlying().set_brackets({}, {});
+@\exposid{underlying_}@.underlying().set_separator(@\exposid{STATICALLY-WIDEN}@<charT>(": "));
+\end{codeblock}
+\end{itemdescr}
+
+\indexlibrarymember{parse}{\exposid{range-default-formatter}}%
+\begin{itemdecl}
+template<class ParseContext>
+  constexpr typename ParseContext::iterator
+    parse(ParseContext& ctx);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+Equivalent to: \tcode{return \exposid{underlying_}.parse(ctx);}
+\end{itemdescr}
+
+\indexlibrarymember{format}{\exposid{range-default-formatter}}%
+\begin{itemdecl}
+template<class FormatContext>
+  typename FormatContext::iterator
+    format(@\exposid{maybe-const-map}@& r, FormatContext& ctx) const;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+Equivalent to: \tcode{return \exposid{underlying_}.format(r, ctx);}
+\end{itemdescr}
+
+\indexlibraryglobal{\exposid{range-default-formatter}}%
+\begin{codeblock}
+namespace std {
+  template<ranges::@\libconcept{input_range}@ R, class charT>
+  struct @\exposid{range-default-formatter}@<range_format::set, R, charT> {
+  private:
+    using @\exposid{maybe-const-set}@ = @\exposid{fmt-maybe-const}@<R, charT>;                          // \expos
+    range_formatter<remove_cvref_t<ranges::range_reference_t<maybe-const-set>>,
+                    charT> @\exposid{underlying_}@;                                         // \expos
```

Comment:
```suggestion
    using @\exposid{maybe-const-set}@ = @\exposid{fmt-maybe-const}@<R, charT>;                            // \expos
    range_formatter<remove_cvref_t<ranges::range_reference_t<@\exposid{maybe-const-set}@>>,   // \expos
                    charT> @\exposid{underlying_}@;
```

Reply:
Do we really want this change? The // \expos is currently on the same line as the identifier it applies to, which seems preferable to me

Reply:
Yikes! Sorry, I thought the identifier was on the previous line.